### DISCUSSION TITLE
Simple custom message subscriber

### DISCUSSION THREAD

**Discussion by kinrep71** [noelex/rclnet#3] [Q&A] [answered]
Hi all,

I am a ROS2 beginner and I would like use a rcl.net to communicate with ROS2 nodes via publishers and subscribers in my C# application. I have a ROS2 foxy x64 distribution running in Windows. 

I just tried to start with simple ROS2 foxy publisher and rcl.net subscriber of Std.String message. However it seems that my ROS2 distribution is not detected by rcl.net. Do you know how to attach the ROS2 distribution to C# rcl.net project? Thank you in advance
![image](https://user-images.githubusercontent.com/49516152/229998884-8658a30d-f610-4a90-b388-19b1a1f4e439.png)
![image](https://user-images.githubusercontent.com/49516152/229999057-588a38a8-8022-4327-91c1-b173a9e12f43.png)

**Comment by noelex** [comment, marked as answer]
You need to source `setup.bat` of the ROS distro you wish to use before running the application.

If you would like to debug the project in Visual Stuidio, you can modify `launchSettings.json` to manually setup ROS environment variables. 
 
For example:
```json
{
  "profiles": {
    "Foxy": {
      "commandName": "Project",
      "environmentVariables": {
        "ROS_DISTRO": "foxy",
        "ROS_LOCALHOST_ONLY": "0",
        "ROS_PYTHON_VERSION": "3",
        "ROS_VERSION": "2",
        "PATH": "D:/sdk/ros/foxy/bin;C:/Program Files/OpenSSL-Win64/bin;C:/ProgramData/chocolatey/lib/tinyxml2/lib",
        "AMENT_PREFIX_PATH": "D:/sdk/ros/foxy"
      }
    }
  }
}
```
**Note: Change the paths to match your system's setup.**

**Reply by kinrep71** [reply]
The simple rcl.net subscriber is able to receive a data from ROS2 foxy publisher now! Thank you so much :) 
![image](https://user-images.githubusercontent.com/49516152/230027202-29c9e85f-4b66-40fd-ac68-f105fcc9b205.png)
